src/SDKs/Batch/DataPlane/Azure.Batch.Unit.Tests/BatchSharedKeyCredentialUnitTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Azure.Batch.Unit.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Batch.Protocol;
+    using Microsoft.Net.Http.Server;
+    using Xunit;
+
+    public class BatchSharedKeyCredentialUnitTests
+    {
+        private const string url = "http://localhost:2055";
+
+        [Theory]
+        [MemberData(nameof(HttpMethods))]
+        public async Task TestSharedKeySigning_ContentLength(HttpMethod httpMethod)
```

Comment:
Normal convention is not to add Async suffix to tests - they are leaf methods and never called by other places. I'm unsure if there is other prior art in this project though.